yt_dlp/extractor/chaturbate.py

```diff
@@ -41,10 +42,54 @@ class ChaturbateIE(InfoExtractor):
     }]
 
     _ROOM_OFFLINE = 'Room is currently offline'
+    _ROOM_PRIVATE = 'Room is currently in a private show'
+    _ROOM_AWAY = 'Performer is currently away'
+    _ROOM_PASSWORD = 'Room is password protected'
+    _ROOM_HIDDEN = 'Hidden session in progress'
+    _ROOM_BLOCKED = 'Room is not avaiable in this region'
 
-    def _real_extract(self, url):
-        video_id, tld = self._match_valid_url(url).group('id', 'tld')
+    def _extract_from_api(self, video_id, tld):
+        req = Request(
+            f'https://chaturbate.{tld}/get_edge_hls_url_ajax/',
+            data=f'room_slug={video_id}'.encode(),
+            headers={
+                'X-Requested-With': 'XMLHttpRequest',
+                'Content-Type': 'application/x-www-form-urlencoded',
+            },
+        )
+        response = self._download_json(req, video_id, fatal=False)
+
+        status = response.get('room_status')
+        if status == 'offline':
+            raise ExtractorError(self._ROOM_OFFLINE, expected=True)
+        elif status == 'private':
+            raise ExtractorError(self._ROOM_PRIVATE, expected=True)
+        elif status == 'away':
+            raise ExtractorError(self._ROOM_AWAY, expected=True)
+        elif status == 'hidden':
+            raise ExtractorError(self._ROOM_HIDDEN, expected=True)
+        elif status == 'password protected':
+            raise ExtractorError(self._ROOM_PASSWORD, expected=True)
+        elif status != 'public':
+            return None
```

Comment:
with the above `_ERROR_MAP` suggestion applied, we can simplify this:
```suggestion
        if status != 'public':
            if error := self._ERROR_MAP.get(status):
                raise ExtractorError(error, expected=True)
            self.report_warning('Falling back to webpage extraction')
            return None
```